### PR TITLE
Add prop to scroll to top when selecting a facet

### PR DESCRIPTION
**Pull request by thalytafabrine**
#### What problem is this solving?

With this prop you can choose whether the page should scroll to the top when selecting a facet

#### How to test it?

[prop with value true](https://thalyta--eriksbikeshop.myvtex.com/bike?_q=bike&map=ft)
[without the prop](https://testscroll--equishopper.myvtex.com/shirt?_q=shirt&map=ft)

#### Screenshots or example usage:

scrollprop

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -284,6 +284,7 @@ Check out the [**Product Summary documentation**](https://vtex.io/docs/component
 | `maxItemsCategory`   | `number`                 | Maximum number of category items to be displayed before the See More button is triggered.     | `8`             |
 | `initiallyCollapsed` | `Boolean` | Makes the search filters start out collapsed (`true`) or open (`false`). | `false` |
 | `filtersTitleHtmlTag` | `string` | HTML tag for the filter's title. | `h5` |
+| `scrollToTop` | `Boolean` | Scrolls the page to the top (`true`) or not (`false`) when selecting a facet. | `false` |
```
Comment (klzns) [inline docs/README.md:287]:
```suggestion
| `scrollToTop` | `enum` | Scrolls the page to the top (`auto` or `smooth`) or not (`none`) when selecting a facet. | `none` |
```

File: react/FilterNavigator.js [hunk 2]
```diff
@@ -71,6 +71,7 @@ const FilterNavigator = ({
   maxItemsDepartment = 8,
   maxItemsCategory = 8,
   filtersTitleHtmlTag = 'h5',
+  scrollToTop = false,
```
Comment (klzns) [inline react/FilterNavigator.js:74]:
```suggestion
  scrollToTop = 'none',
```

File: react/hooks/useFacetNavigation.js [hunk 3]
```diff
@@ -175,7 +175,7 @@ export const buildNewQueryMap = (
   return buildQueryAndMap(querySegments, mapSegments, facets, selectedFacets)
 }
 
-const useFacetNavigation = selectedFacets => {
+const useFacetNavigation = (selectedFacets, scrollToTop = false) => {
```
Comment (klzns) [inline react/hooks/useFacetNavigation.js:178]:
```suggestion
const useFacetNavigation = (selectedFacets, scrollToTop = 'none') => {
```

File: react/hooks/useFacetNavigation.js [hunk 4]
```diff
@@ -191,6 +191,10 @@ const useFacetNavigation = selectedFacets => {
         selectedFacets
       )
 
+      if (scrollToTop) {
```
Comment (klzns) [inline react/hooks/useFacetNavigation.js:194]:
```suggestion
      if (scrollToTop !== 'none') {
```

File: react/hooks/useFacetNavigation.js [hunk 5]
```diff
@@ -191,6 +191,10 @@ const useFacetNavigation = selectedFacets => {
...
+        window.scrollTo(0, 0)
```
Comment (klzns) [inline react/hooks/useFacetNavigation.js:195]: Use the smooth-scroll polyfill http://iamdustan.com/smoothscroll/

Then we can use `behavior: 'smooth'`

Comment (thalytafabrine) [reply react/hooks/useFacetNavigation.js]: fixed! I used `smooth` in [this workspace](https://thalyta--eriksbikeshop.myvtex.com/bike?_q=bike&map=ft). Thank you!!

**Pull request merged**